enhancements/78-dynamically-reclaiming-resources/README.md

```diff
@@ -0,0 +1,122 @@
+# Dynamically reclaim resources of Pods of a Workload
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+    - [Risks and Mitigations](#risks-and-mitigations)
+- [API](#api)
+- [Graduation Criteria](#graduation-criteria)
+- [Testing Plan](#testing-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+    - [E2E tests](#e2e-tests)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+The resources of pods of a Workload are reclaimed by kueue after successful completion of the Pod. 
+This is beneficial for workloads which cannot proceed due to the unavailability 
+of the little resources for its corresponding pods and Workload does not get 
+admitted.
+
+Here after, *pods of a Workload* are synonymously used to refer *pods of a Job* and vice-versa because, 
+Job manages the Workload and the pods associated with Job are associated with the Workload. 
+
+
+## Motivation
+
+Currently, the resources owned by a Job are reclaimed by kueue only when the whole Job finishes.
+For jobs with multiple pods, the resources will be reclaimed only after the last Pod
+of the Job finishes. This is not efficient as the pods of parallel Job may have laggards
+consuming the unused resources of the jobs those are currently executing.
+
+### Goals
+
+- Utilize the unused resources of the Job which running in kueue.
+
+### Non-Goals
+
+- A running Job on queue should not be preempted.
+- The resources that are being used by a Job should not be allocated to any other Job.
+
+## Proposal
+
+Reclaim the resources of the succeeded pods of a Job as soon as it completes its execution.
```

Comment:
Some cases I didn't met but maybe worth considering: when job failed, all the pods should be restarted. If so, reclaiming the resources of succeed pods is not expected. Then we should provide an optional field for end-users to choose, like define a `reclaimPolicy` in workload, default to non-reclaim.

Reply:
Explained pod failure scenarios w.r.t fields restartPolicy, backoffLimit and activeDeadlineSeconds. But, as proposed, reclaiming of failed resources will not be done unless the job gets completed as either Failed or Succeeded.

Reply:
Let me explain this clearly: when running some kinds of workloads(e.g. HPC), we can't tolerate pod failure, then we'll restart all the pods, at that moment, if some pods got succeed and we reclaimed their resources, with some probabilities, these pods may failed in creation for insufficient resources. 

But I'm ok with current implementations for a startup until we come up with more user cases.